### PR TITLE
Upload HTML report as artifact

### LINKED ISSUE #388: Upload html report as artifact on loaner tasks
Depends on https://github.com/mozilla/code-review/issues/439

That way the report would be automatically viewable from the S3 artifacts

### PR DESCRIPTION
**Pull request by La0**
Fixes #388

### CODE REVIEW

**Comment by La0** [comment]
Artifact published on the CI task : https://community-tc.services.mozilla.com/tasks/RNk956E9QWiaO_zjGaQKig#artifacts

File: report/requirements.txt [hunk 1]
```diff
@@ -1 +1,4 @@
-taskcluster==24.2.0
+# Switch to next release of Taskcluster once https://github.com/taskcluster/taskcluster/pull/2294 lands
+https://github.com/La0/taskcluster/archive/client-py-upload-artifact.zip#egg=taskcluster&subdirectory=clients/client-py
```
Comment (marco-c) [inline report/requirements.txt]: We could actually wait for that before merging this, so we don't have to do this (and we don't have to do the changes in setup.py).

File: report/firefox_code_coverage/codecoverage.py [hunk 2]
```diff
@@ -310,6 +313,27 @@ def download_grcov():
...
+    assert not base_artifact.endswith("/"), "No trailing / in base_artifact"
+
+    # Use Taskcluster proxy when available
+    taskcluster.auth()
+
+    for path in Path(report_dir).rglob("*"):
+
+        filename = str(path.relative_to(report_dir))
+        content_type = magic.from_file(str(path), mime=True)
+        print("Uploading {} as {}".format(filename, content_type))
```
Comment (marco-c) [inline report/firefox_code_coverage/codecoverage.py]: logger.debug, or this will print a lot of stuff

**Comment by La0** [comment]
Yeap, i'll wait for the taskcluster changes to land before merging this.

File: report/requirements.txt (hunk 1 above)
Comment (La0) [reply report/requirements.txt]: Updated to 24.3.1

File: report/firefox_code_coverage/codecoverage.py (hunk 2 above)
Comment (La0) [reply report/firefox_code_coverage/codecoverage.py]: Done

**Pull request merged**